src/performance/manager.py

```
@@ -0,0 +1,182 @@
+                "optimization_count": len(self.optimizer.get_optimization_history())
+            }
+        }
+        
+    def analyze_trends(self, days: int = 7) -> Dict[str, Any]:
+        """Analyze performance trends over specified period"""
+        cutoff = datetime.now() - timedelta(days=days)
+        history = self.monitor.get_history()
+        recent_history = [m for m in history if m["timestamp"] >= cutoff]
+        
+        trends = {}
+        operations = set(m["operation"] for m in recent_history)
```

Ruff (0.8.2)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)